pages/blog/css-with-react.mdx

```diff
@@ -0,0 +1,274 @@
+</HomeContainerWrapper>
+
+// Good
+<Box
+  padding={3}
+  fontWeight='bold'
+  color='white'
+  bg='blue'
+>
+  Hello
+</Box>
+```
+
+Now, I know what you're thinking. Inline styles 🤮. I'll admit, I wasn't sold at first. But it grew on me. And that brings us to now.
+
+## Current State
+
+I've recently been working with [CSS Modules](https://github.com/css-modules/css-modules) and [Tailwind CSS](https://tailwindcss.com/).
+Both of these approaches to styling React applications have prompted me to step back and evaluate the ecosystem.
+
+I've tried to summarize and collect pros and cons for various approaches of styling your React application. The choice that's best for you
+likely depends on:
+
+- Your experience (stick with what you know)
+- Your team
+- The size of your application
+- What you're trying to build
+
+## Pros and Cons
+
+### Vanilla CSS
+
+import ProsCard from '../../components/ProsCard';
+import ConsCard from '../../components/ConsCard';
+
+<ProsCard
+  title="vanilla CSS"
+  pros={['No toolchain necessary. Works with every browser & language.']}
+/>
+
+<ConsCard
+  title="vanilla CSS"
+  cons={['Not a viable option for large web applications.']}
+/>
+
+### CSS Modules
+
+[CSS Modules](https://github.com/css-modules/css-modules) would be my choice for smaller teams that don't need to share
+components across applications.
+
+[Next.js](https://nextjs.org) has support for CSS Modules, which means you don't have to worry about setting up [Webpack]().
+It also sets up [autoprefixer](https://css-tricks.com/autoprefixer/), which makes CSS Modules an attractive alternative to CSS-in-JS in some cases.
+
+<ProsCard
+  title="CSS modules"
+  pros={[
+    'Class names are scoped to the corresponding component.',
+    'CSS variables allow you to reuse values.',
+    'Your application can work without JavaScript.',
+    'Lowest barrier to entry.',
+    'Works with any language.'
+  ]}
+/>
+
+<ConsCard
+  title="CSS modules"
+  cons={[
+    'Requires confguring a toolchain like Webpack.',
+    'Distribution is hard.',
+    'Lack of nesting and CSS linting.',
+    `Some argue there's more context switching between different files`
+  ]}
+/>
+
+### CSS-in-JS
+
+There's many different CSS-in-JS libraries. The two most popular are [styled-components](https://styled-components.com/) and [Emotion](https://emotion.sh/docs/introduction).
+
+There's also zero-runtime solutions like [Linaria](https://github.com/callstack/linaria), where CSS is extracted to CSS files at build-time.
+To see comparisions between different libraries, see [CSS-in-JS benchmarks.](https://github.com/A-gambit/CSS-IN-JS-Benchmarks)
+
+<ProsCard
+  title="CSS-in-JS"
+  pros={[
+    'Easy to distribute as reusable components on NPM.',
+    'You can take advatange of anything in the JavaScript ecosystem.',
+    'Styles are colocated with components, making it easy to delete code.',
+    'Easy to provide multiple themes.'
+  ]}
+/>
+
+<ConsCard
+  title="CSS-in-JS"
+  cons={[
```

Comment:
I'd add naming is hard here too since you need to name those component :P Also they use context extensively (in contrast css + custom properties are much faster)